### PR TITLE
RAC-785: Simple select option replacement

### PR DESCRIPTION
**Pull request by pierrejolly**
**Description (for Contributor and Core Developer)**

- Adding id on SwitcherButton to improve accessibility and link the label to the button
- Adding SQL SearchAttributeOptions query in AttributeOption public API

### CODE REVIEW

File: src/Akeneo/Pim/Structure/Bundle/Query/PublicApi/AttributeOption/Sql/SqlSearchAttributeOptions.php [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+            'locale_code' => $searchParameters->getLocale(),
+            'include_codes' => $searchParameters->getIncludeCodes(),
+            'exclude_codes' => $searchParameters->getExcludeCodes(),
+        ], [
+            'attribute_id' => \PDO::PARAM_STR,
+            'search' => \PDO::PARAM_STR,
+            'locale_code' => \PDO::PARAM_STR,
+            'include_codes' => Connection::PARAM_STR_ARRAY,
+            'exclude_codes' => Connection::PARAM_STR_ARRAY,
+        ])->fetchColumn();
```
Comment (SamirBoulil) [inline src/Akeneo/Pim/Structure/Bundle/Query/PublicApi/AttributeOption/Sql/SqlSearchAttributeOptions.php:62]: Can you isolate this chunk into a private ?

File: src/Akeneo/Pim/Structure/Bundle/Query/PublicApi/AttributeOption/Sql/SqlSearchAttributeOptions.php [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+            'offset' => $searchParameters->getOffset(),
+        ], [
+            'attribute_code' => \PDO::PARAM_STR,
+            'search' => \PDO::PARAM_STR,
+            'locale_code' => \PDO::PARAM_STR,
+            'include_codes' => Connection::PARAM_STR_ARRAY,
+            'exclude_codes' => Connection::PARAM_STR_ARRAY,
+            'limit' => \PDO::PARAM_INT,
+            'offset' => \PDO::PARAM_INT,
+        ])->fetchAll(\PDO::FETCH_ASSOC);
```
Comment (SamirBoulil) [inline src/Akeneo/Pim/Structure/Bundle/Query/PublicApi/AttributeOption/Sql/SqlSearchAttributeOptions.php:106]: Same here :)

File: src/Akeneo/Pim/Structure/Component/Query/PublicApi/AttributeOption/SearchAttributeOptionsResult.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+class SearchAttributeOptionsResult
+{
+    private int $matchesCount;
+
+    /** @var AttributeOption[] */
+    private array $items;
+
+    public function __construct(array $items, int $matchesCount)
+    {
+        $this->items = $items;
```
Comment (SamirBoulil) [inline src/Akeneo/Pim/Structure/Component/Query/PublicApi/AttributeOption/SearchAttributeOptionsResult.php:20]: I wonder if we should: assert allInstanceOfAttributeOption ?

**Review by SamirBoulil — APPROVED** [review]
GTM after fix

**Review by ghoussard — APPROVED** [review]

File: src/Akeneo/Pim/Structure/Bundle/Query/PublicApi/AttributeOption/Sql/SqlSearchAttributeOptions.php [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+use Akeneo\Pim\Structure\Component\Query\PublicApi\AttributeOption\AttributeOption;
+use Akeneo\Pim\Structure\Component\Query\PublicApi\AttributeOption\SearchAttributeOptionsInterface;
+use Akeneo\Pim\Structure\Component\Query\PublicApi\AttributeOption\SearchAttributeOptionsParameters;
+use Akeneo\Pim\Structure\Component\Query\PublicApi\AttributeOption\SearchAttributeOptionsResult;
+use Doctrine\DBAL\Connection;
+
+/**
+ * @copyright 2021 Akeneo SAS (https://www.akeneo.com)
+ * @license http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
```
Comment (StevenVAIDIE) [inline src/Akeneo/Pim/Structure/Bundle/Query/PublicApi/AttributeOption/Sql/SqlSearchAttributeOptions.php:15]: :thinking: 
```suggestion
 * @license https://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
```

File: src/Akeneo/Pim/Structure/Bundle/Query/PublicApi/AttributeOption/Sql/SqlSearchAttributeOptions.php [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+SQL;
+
+        $matchesCount = $this->connection->executeQuery($sql, [
+            'attribute_code' => $attributeCode,
+            'search' => sprintf('%%%s%%', $searchParameters->getSearch() ?? ''),
+            'locale_code' => $searchParameters->getLocale(),
+            'include_codes' => $searchParameters->getIncludeCodes(),
+            'exclude_codes' => $searchParameters->getExcludeCodes(),
+        ], [
+            'attribute_id' => \PDO::PARAM_STR,
```
Comment (StevenVAIDIE) [inline src/Akeneo/Pim/Structure/Bundle/Query/PublicApi/AttributeOption/Sql/SqlSearchAttributeOptions.php:131]:
```suggestion
            'attribute_code' => \PDO::PARAM_STR,
```

**Review by StevenVAIDIE — APPROVED** [review]

**Review by juliensnz — APPROVED** [review]

**Pull request merged**